### PR TITLE
Shut down the application's event loop when it gets deallocated.

### PR DESCRIPTION
**Pull request by MrMage**
Without this, every time we initialize an `Application`, it leaks an event loop thread. This probably won't happen in production, but it might happen in tests, where each test case creates its own `Application` instance for testing.

### CODE REVIEW

File: Sources/Vapor/Application.swift [hunk 1]
```diff
@@ -24,6 +24,9 @@ public final class Application: Container {
     /// See ServiceCacheable.serviceCache
     public let serviceCache: ServiceCache
 
+    /// The event loop group that we derive the event loop below from, so we can close it in `deinit`
+    private var eventLoopGroup: EventLoopGroup = MultiThreadedEventLoopGroup(numThreads: 1)
```
Comment (tanner0101) [inline Sources/Vapor/Application.swift]: please set this property in the initializer.

File: Sources/Vapor/Application.swift [hunk 2]
```diff
@@ -24,6 +24,9 @@ public final class Application: Container {
...
+
     /// See Worker.eventLoop
     public var eventLoop: EventLoop
```
Comment (tanner0101) [inline Sources/Vapor/Application.swift]: this can be a computed var.

File: Sources/Vapor/Application.swift [hunk 3]
```diff
@@ -60,6 +63,14 @@ public final class Application: Container {
             log.info("Compile your application with `-c release` to enable code optimizations.")
         }
     }
+    
+    deinit {
+        eventLoopGroup.shutdownGracefully {
+            if let error = $0 {
+                print("[Vapor] shutting down app event loop: \(error)")
```
Comment (tanner0101) [inline Sources/Vapor/Application.swift]: you can use the global `ERROR` function

**Comment by MrMage** [comment]
Please take another look.

**Pull request merged**